extensions/sql-database-projects/src/models/project.ts

```diff
@@ -928,21 +928,27 @@ export class Project {
 
 	/**
 	 * Adds the list of sql files and directories to the project, and saves the project file
-	 * @param absolutePath Absolute path of the folder
+	 * @param list list of files and folder Uris. Files and folders must already exist. No files or folders will be added if any do not exist.
 	 */
-	public async addToProject(list: string[]): Promise<void> {
+	public async addToProject(list: Uri[]): Promise<void> {
+		// verify all files/folders exist. If not all exist, none will be added
+		for (let file of list) {
+			const exists = await utils.exists(file.fsPath);
 
-		for (let i = 0; i < list.length; i++) {
-			let file: string = list[i];
-			const relativePath = utils.trimChars(utils.trimUri(Uri.file(this.projectFilePath), Uri.file(file)), '/');
+			if (!exists) {
+				throw new Error(constants.fileOrFolderDoesNotExist(file.fsPath));
+			}
+		}
+
+		for (let file of list) {
```

Comment:
Does it make sense to merge this and the for loop above to check the existence as well as do the remaining processing? We can cleanup script and folder items, if it is half processed.
(I am fine either way though :-) )

Reply:
I prefer checking before over having to go back and cleanup, in case something gets missed